pkg/cloudprovider/providers/vsphere/vclib/diskmanagers/vmdm.go

```diff
@@ -13,13 +13,37 @@ import (
 // vmDiskManager implements VirtualDiskProvider interface for creating volume using Virtual Machine Reconfigure approach
 type vmDiskManager struct {
 	diskPath      string
-	volumeOptions vclib.VolumeOptions
-	vmOptions     vclib.VMOptions
+	volumeOptions *vclib.VolumeOptions
+	vmOptions     *vclib.VMOptions
 }
 
 // Create implements Disk's Create interface
 // Contains implementation of VM based Provisioning to provision disk with SPBM Policy or VSANStorageProfileData
 func (vmdisk vmDiskManager) Create(ctx context.Context, datastore *vclib.Datastore) (err error) {
+	if vmdisk.volumeOptions.StoragePolicyName != "" {
+		pbmClient, err := vclib.NewPbmClient(ctx, datastore.Client())
+		if err != nil {
+			glog.Errorf("Error Occurred while creating new pbmClient, err: %+v", err)
+			return err
+		}
+		vmdisk.volumeOptions.StoragePolicyID, err = pbmClient.ProfileIDByName(ctx, vmdisk.volumeOptions.StoragePolicyName)
```

Comment:
Kept StoragePolicyID, as this is required for Attach Operation. As we discussed added validation, to make sure both are not passed for volume provisioning.